### PR TITLE
Fuzz v3

### PR DESCRIPTION
**Pull request by catenacyber**
Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/2859

Describe changes:
- Adds three fuzz targets
- Change compilation scheme so as to compile fuzz targets (option --enable-fuzztargets for configure)

Modifies #3835 :
- adds fuzz target for `SigInit` (rule parsing)
- Adds a README
- Adds onefile.c simple driver
- Code style fixes as reviewed

I think that this PR is not complete but I would like some feedback, that is why I am doing it.
I kept the history for commits so as to follow my logic.

### CODE REVIEW

File: src/tests/fuzz/fuzz_applayerparserparse.c [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+
+#include "suricata-common.h"
+#include "app-layer-detect-proto.h"
+#include "flow-util.h"
+#include "app-layer-parser.h"
+
+#define HEADER_LEN 6
+
+void fuzz_openFile(const char * name) {
```
Comment (SirkoHoeer) [inline src/tests/fuzz/fuzz_applayerparserparse.c]: this function is not necessary.

File: src/tests/fuzz/fuzz_applayerparserparse.c [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+    Flow * f;
+    TcpSession ssn;
+
+    if (size < HEADER_LEN) {
+        return 0;
+    }
+
+    if (alp_tctx == NULL) {
```
Comment (SirkoHoeer) [inline src/tests/fuzz/fuzz_applayerparserparse.c]: you could use an init function i.e. int LLVMFuzzerInitialize(int *argc, char ***argv)

Comment (catenacyber) [reply src/tests/fuzz/fuzz_applayerparserparse.c]: Yes but that is not recommended by libfuzzer : http://llvm.org/docs/LibFuzzer.html#startup-initialization

File: src/tests/fuzz/fuzz_applayerparserparse.c (hunk 1 above)
Comment (catenacyber) [reply src/tests/fuzz/fuzz_applayerparserparse.c]: Indeed, it is not used yet.
In other projects, I use this function to open a file for logging.

How do you achieve this ?

File: src/tests/fuzz/fuzz_applayerparserparse.c (hunk 2 above)
Comment (SirkoHoeer) [reply src/tests/fuzz/fuzz_applayerparserparse.c]: Ok, you are right. I prefer this kind of solution

File: src/tests/fuzz/README [hunk 3]
```diff
@@ -0,0 +1,11 @@
+How to run fuzzing ?
```
Comment (victorjulien) [inline src/tests/fuzz/README]: how do I control which Suricata repo+branch is used?

Comment (catenacyber) [reply src/tests/fuzz/README]: This is in `Dockerfile` used by oss-fuzz
See https://github.com/google/oss-fuzz/pull/2533/files#diff-f132db5f0e91b1506efa943dd4eb789bR30

**Pull request closed without merging**